include/nbl/builtin/hlsl/cmath.hlsl

```diff
@@ -99,7 +116,7 @@ int32_t ilogb(T x)
     using uint_type = typename nbl::hlsl::numeric_limits<T>::uint_type;
     const int32_t shift = (impl::num_base<T>::float_digits-1);
     const uint_type mask = ~(((uint_type(1) << shift) - 1) | (uint_type(1)<<(sizeof(T)*8-1)));
-    int32_t bits = (bit_cast<uint_type, T>(x) & mask) >> shift;
+    int32_t bits = (std::bit_cast<uint_type, T>(x) & mask) >> shift; //nbl::hlsl::bit_cast sucks as it pulls the ambiguous one from intrinsics.h
```

Comment:
explain

Reply:
I want all this to be tested and compile as both C++ and HLSL, you obviously haven't tested